internal/credential/vault/vault.go

```diff
@@ -0,0 +1,242 @@
+package vault
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/go-rootcerts"
+	vault "github.com/hashicorp/vault/api"
+	"github.com/mitchellh/mapstructure"
+)
+
+type clientConfig struct {
+	Addr          string
+	Token         TokenSecret
+	CaCert        []byte
+	ClientCert    []byte
+	ClientKey     KeySecret
+	TlsServerName string
+	TlsSkipVerify bool
+	Namespace     string
+}
+
+func (c *clientConfig) isValid() bool {
+	if c == nil || c.Addr == "" || len(c.Token) < 0 {
+		return false
+	}
+	return true
+}
+
+func (c *clientConfig) isClientTLS() bool {
+	if len(c.ClientCert) > 0 && len(c.ClientKey) > 0 {
+		return true
+	}
+	return false
+}
+
+type client struct {
+	cl    *vault.Client
+	token TokenSecret
+}
+
+func newClient(c *clientConfig) (*client, error) {
+	const op = "vault.newClient"
+	if !c.isValid() {
+		return nil, errors.New(errors.InvalidParameter, op, "invalid configuration")
+	}
+	vc := vault.DefaultConfig()
```

Comment:
Is there a reason below not to call `ConfigureTLS` on the config object, instead of duplicating a bunch of its stuff below? Are you trying to avoid using env vars? @mgaffney

Reply:
`ConfigureTLS` reads the client cert and the client cert key from files which is a problem for us since we are getting those from the database.